Source/Cell/Cell.cpp

```diff
@@ -85,3 +115,11 @@ bool Cell::IsValid()
 {
 	return is_valid;
 }
+bool Cell::IsLegalMove()
```

Comment:
@arive298  Im not sure if we will do the check move here. I think we can do this at the continent level.